doc/continuous_integration/dockerhub.md

```diff
@@ -0,0 +1,57 @@
+
+* [Navigation2 on DockerHub](https://hub.docker.com/r/rosplanning/navigation2)
+
+While DockerHub does not require the use of configuration files in the source repo, the Dockerfiles and respective scripts used to customize automated builds of images are tracked in the [`.dockerhub`](/.dockerhub) directory. These scripts include custom build phase hooks used by DockerHub during automated builds. More info on automated builds can be found here:
+
+* [How Automated Builds work](https://docs.docker.com/docker-hub/builds)
+* [Advanced options for Autobuild and Autotest](https://docs.docker.com/docker-hub/builds/advanced)
+
+Automated Builds must be enabled via the build configurations menu within the repo's administrative console on DockerHub. For reference, a figure of the project's build configurations is shown here:
+
+![DockerHub Build Configurations](figs/dockerhub_build_configurations.png)
+
+Here the repo's source repository is linked to the project's GitHub repo, while Autotest is disabled given a dedicated CI service is separately used to run test jobs. Repository linking is enabled so that whenever the base image is updated on DockerHub, it will also trigger a build for the project's DockerHub repository. Note that this only works for non-official images. Two build rules are added for the master branch, providing both a release and debug tag for CI to pull from. The paths to the relative Dockerfiles are designated, while the build context is intentionally left empty. This ensures that the build phase hooks within the same paths are used appropriately. Build caching is also enabled to shorten image turnaround time if multiple rebuilds a day are triggered. 
+
+The build hooks, e.g. [build](/.dockerhub/debug/hooks/build) , serve a few purposes. The first being to customize docker commands to override build arguments denoted in the Dockerfile via the `ARG` directives, such as adjusting the colcon mixins for each workspace to enable code coverage, or disabling fail on build failure so the breakages in master don't block the building of CI images for master.
+
+The build hooks secondly allow for the redirection of target Dockerfile and build context, given the use of `ARG` to parameterize the parent image to build `FROM` inhibits the normal use of repository links. This is a limitation of the Dockerfile parsing used by DockerHub to infer what repo/tag the build rule should trigger from. 
+
+As a workaround, the build rule points to a dummy Docker file that includes the desired parent image tag to build `FROM`, while the build hook then uses the main Dockerfile with the same `FROM_IMAGE`. This allows the slower debug CI to build and test only using the default RMW, while allowing the faster release CI jobs to build and test from an image with more RMWs installed.
```

Comment:
I don't think this explains the "dummy dockerfile" thing

Reply:
We no longer use this complex workaround. Instead, we now trigger dockerhub builds using it's API via cron job rather than using the Autobuild setting. This has been now been simplified.